cli/api/dbadapters/bigquery.ts

```diff
@@ -330,72 +330,79 @@ export class BigQueryDbAdapter implements IDbAdapter {
     return retry(
       async () => {
         try {
-          const job = await this.getClient().createQueryJob({
-            useLegacySql: false,
-            jobPrefix: "dataform-" + (jobPrefix ? `${jobPrefix}-` : ""),
-            query,
-            params,
-            labels,
-            location,
-            dryRun
-          });
-          const resultStream = job[0].getQueryResultsStream();
-          return new Promise<IExecutionResult>((resolve, reject) => {
-            if (isCancelled) {
-              resultStream.end();
-              reject(new Error("Query cancelled."));
-              return;
-            }
-            onCancel?.(() => {
-              resultStream.end();
-              reject(new Error("Query cancelled."));
+          if (dryRun && query.includes('-- @dry-run:skip-results-check')) {
```

Comment:
I'm not a fan of making this depend on the SQL contents, because we have parameters that can be passed around instead - also this won't be backwards compatible with older core versions.

I think if you look upstream, we should just be able to skip the second statement of assertions here https://github.com/dataform-co/dataform/blob/6f13b4a283805928e0233004b50e8a074c90cd33/cli/api/commands/run.ts#L416 or one level higher up. This way inserting to the SQL during compilation won't be needed